src/app/clusters/level-control/level-control.cpp

```diff
@@ -1122,6 +1146,10 @@ static bool areStartUpLevelControlServerAttributesNonVolatile(EndpointId endpoin
 }
 #endif // IGNORE_LEVEL_CONTROL_CLUSTER_START_UP_CURRENT_LEVEL
 
-void emberAfPluginLevelControlClusterServerPostInitCallback(EndpointId endpoint) {}
+void emberAfPluginLevelControlClusterServerPostInitCallback(EndpointId endpoint)
+{
+    VerifyOrDie(endpoint < FIXED_ENDPOINT_COUNT);
+    events[endpoint] = { .isActive = false, .callback = emberAfLevelControlClusterServerTickCallback, .endpoint = endpoint };
```

Comment:
I think we can do one better, and avoid pools entirely.

Because the callback is always fixed, and the only variable argument here is the endpoint ID, we can actually stuff the endpoint ID in the context for `StartTimer` AND `CancelTimer`. This way, we can avoid the need to have an `isActive` flag, and just call `CancelTimer` (which will ensure to cancel the timer matching the combination of both the handler + endpoint).

No more pools! :)

Reply:
The `context` is a `void *` pointer, but I don't see why we cannot stuff a non-pointer value in there. We certainly have at least a pointer width's worth of storage in there, and since `EndpointId` type is just 16-bits, we should be safe.

Reply:
That's a good point. There's one problem I see. In the `schedule` function, `EndpointId endpoint` is passed as value. But `DeviceLayer::SystemLayer().StartTimer` requires we pass a ptr to the context `void* appState`.

I can't pass `endpoint` as a ptr because this would be on the stack of the caller function, and immediately would be invalid. I _could_ typecast the endpoint and use the 4bytes that it allocates for a ptr. I'm the one unwrapping the context on the callback. But it isn't very elegant, and I'd be doing a uint8_t -> void* -> uint_t typecasting. What do you guys think?

Reply:
I think casting it is fine, like I mentioned above. Context is meant to be an opaque closure that the timer subsystem itself is not going to do anything with - it just forwards it along.